### ISSUE TITLE
"To run this application, you must install .NET."

### ISSUE THREAD

**Issue by FoxVixen3** [zxcPandora/Titanfall2-SkinTool#12]
I have .NET 5.0 and 6.0 Installed and restarted my PC, tried running it as admin and Still same message. Newest Version (v0.8.3.4)
Pressing the download button did something when I first started it but now it does nothing.

**Comment by zxcPandora** [comment]
> I have .NET 5.0 and 6.0 Installed and restarted my PC, tried running it as admin and Still same message. Newest Version (v0.8.3.4) Pressing the download button did something when I first started it but now it does nothing.

you need "desktop app" version in this page:https://dotnet.microsoft.com/en-us/download/dotnet/6.0/runtime

**Comment by FoxVixen3** [comment]
I have, but as said problems still occur.

**Comment by EM4Volts** [comment]
same issue, .net 6 and 4.8 installed but error still persists.

**Comment by hqo998** [comment]
Following up, applied previously listed steps and error persists. This error only occurs on the SkinTool-Maker version and not the SkinTool-User version.
![Screenshot 2023-05-28 1714383](https://github.com/zxcPandora/Titanfall2-SkinTool/assets/6234805/5dc97df6-49ce-40e4-a1ca-21e7d708f62b)
Pressing the Yes button closes the tab with no further changes, same with the no button.

**Comment by NotImportantAtA11** [comment]
Yep, still a problem. I've downloaded like 3 different versions of .NET including runtime, basic, and the desktop version. I've also tried using older SkinTool versions, and it hasn't worked either

**Comment by zxcPandora** [comment]
To be honest, I'm also very strange.
Try this link I found from microsoft?
https://learn.microsoft.com/en-us/dotnet/core/runtime-discovery/troubleshoot-app-launch?view=vs-2022&pivots=os-windows

**Comment by zxcPandora** [comment]
@FoxVixen3 @hqo998 @NotImportantAtA11 @EM4Volts 
I found something maybe can solve it,here's the step:
Add new User **environment variable**
`DOTNET_ROOT=<folder>\dotnet`
`DOTNET_MULTILEVEL_LOOKUP=0`
Add also to your `PATH`
`PATH=%DOTNET_ROOT%;`

Or just like me,add the dotnet path in the `PATH`
![QQ截图20230912114522](https://github.com/zxcPandora/Titanfall2-SkinTool/assets/81985226/42d4fff2-9659-4988-bf18-62521b130df3)

**Comment by NotImportantAtA11** [comment]
It still does not work for me. A new development is that the link in the error message now works, but even after I downloaded that version of .NET it still won't open, and the link will keep opening to the previously installed .NET version. I also edited the paths like you said, and I still get the errors
More than anything I'm just confused on why it isn't recognizing .NET...

**Comment by NotImportantAtA11** [comment]
GUYS I FOUND IT (at least for me)
So i cant remember each specific step (check "Vortex wont recognize .net" its what I used), but the finishing touch was
Open Command Prompt in admin
type:
icacls.exe "C:\Program Files\dotnet" /grant "*S-1-15-2-1:(oi)(ci)(rx)"

icacls.exe "C:\Program Files\dotnet" /grant "*S-1-15-2-2:(oi)(ci)(rx)"

(or the C:\Program Files... stuff is where your .NET is downloaded)

**Comment by zxcPandora** [comment]
> GUYS I FOUND IT (at least for me) So i cant remember each specific step (check "Vortex wont recognize .net" its what I used), but the finishing touch was Open Command Prompt in admin type: icacls.exe "C:\Program Files\dotnet" /grant "*S-1-15-2-1:(oi)(ci)(rx)"
> 
> icacls.exe "C:\Program Files\dotnet" /grant "*S-1-15-2-2:(oi)(ci)(rx)"
> 
> (or the C:\Program Files... stuff is where your .NET is downloaded)

good ,wait for others feedback.
![QQ截图20230914114507](https://github.com/zxcPandora/Titanfall2-SkinTool/assets/81985226/f38f35f0-ff84-468b-a382-f98b4dd75ccc)

**Comment by Finnfx** [comment]
I still have this issue, fix does not work. It will process the files but i still get .net needed error

**Comment by zxcPandora** [comment]
> I still have this issue, fix does not work. It will process the files but i still get .net needed error

Did you see that **NotImportantAtA11**'s solution？

**Comment by suegdu** [comment]
@zxcPandora Facing the same issue here, any recommendations? (I did all the described solutions above and non of them were to be working)

**Comment by Sylive147** [comment]
换了3台电脑都没解决，打上.net SDK 5就好了（高版本并不具有兼容性）
https://dotnet.microsoft.com/en-us/download/dotnet/5.0
